pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentFlushThresholdComputerTest.java

```diff
@@ -105,6 +106,28 @@ public void testUseLastSegmentsThresholdIfSegmentSizeMissing() {
     assertEquals(threshold, segmentSizeThreshold);
   }
 
+  @Test
+  public void testUseLastSegmentsThresholdIfSegmentIsCommittingDueToForceCommit() {
+    long committingSegmentSizeBytes = 500_000L;
+    int committingSegmentSizeThreshold = 25_000;
+    SegmentFlushThresholdComputer computer = new SegmentFlushThresholdComputer();
+
+    CommittingSegmentDescriptor committingSegmentDescriptor = mock(CommittingSegmentDescriptor.class);
+    when(committingSegmentDescriptor.getSegmentSizeBytes()).thenReturn(committingSegmentSizeBytes);
+    when(committingSegmentDescriptor.getStopReason()).thenReturn(REASON_FORCE_COMMIT_MESSAGE_RECEIVED);
```

Comment:
maybe also add a test when the reason is null. (will happen during upgrades if servers are upgraded last)

Reply:
I have actually considered the upgrade scenario, and I believe a unit test is unnecessary in this case. This is because the behavior during the upgrade will be managed in the same way as it was before the upgrade, with the threshold being calculated using much lower values. To put it differently, the miscalculation bug is present both before and during the upgrade. However, after the upgrade, the bug is resolved.